src/Serialization/Address.purs

```diff
@@ -0,0 +1,348 @@
+  :: { network :: Int
+     , paymentStakeCred :: StakeCredentialCsl
+     } -> RewardAddressCsl
+
+foreign import newBaseAddressCsl
+  :: { network :: Int
+     , paymentStakeCred :: StakeCredentialCsl
+     , delegationStakeCred :: StakeCredentialCsl
+     } -> BaseAddressCsl
+
+foreign import addressBytesImpl :: AddressCsl -> ByteArray
+foreign import addressBech32Impl :: AddressCsl -> Bech32String
+
+-- | Ensures that stake credentials are of requested type
+-- therefore allowing to set BaseAddress' type arguments
+foreign import headerCheckBaseAddr
+  :: forall p d
+   . MaybeFfiHelper
+  -> { payment :: CredType p
+     , delegation :: CredType d}
+  -> ByteArray
+  -> BaseAddressCsl
+  -> Maybe (BaseAddress p d)
+
+-- | Ensures that stake credentials are of requested type
+-- therefore allowing to set RewardAddress' type arguments
+foreign import headerCheckRewardAddr
+  :: forall p
+   . MaybeFfiHelper
+  -> { payment :: CredType p}
+  -> ByteArray
+  -> RewardAddressCsl
+  -> Maybe (RewardAddress p)
+
+foreign import baseAddressFromBytesImpl :: MaybeFfiHelper -> ByteArray -> Maybe BaseAddressCsl
+foreign import baseAddressFromBech32Impl :: MaybeFfiHelper -> Bech32String -> Maybe BaseAddressCsl
+
+foreign import rewardAddressFromBytesImpl :: MaybeFfiHelper -> ByteArray -> Maybe RewardAddressCsl
+foreign import rewardAddressFromBech32Impl :: MaybeFfiHelper -> Bech32String -> Maybe RewardAddressCsl
+
+-- | The exported constructor for `BaseAddress`.
+baseAddress
+  :: forall p d
+   . (ToCsl (StakeCred d) StakeCredentialCsl)
+  => (ToCsl (StakeCred p) StakeCredentialCsl)
+  => {network :: NetworkTag, payment :: p, delegation :: d}
+  -> BaseAddress p d
+baseAddress {network, payment, delegation} = BaseAddress
+  { network: network
+  , payment: payment
+  , delegation: delegation
+  }
+
+-- | The exported constructor for `RewardAddress`.
+rewardAddress
+  :: forall p
+   . (ToCsl (StakeCred p) StakeCredentialCsl)
+  => {network :: NetworkTag, payment :: p}
+  -> RewardAddress p
+rewardAddress {network, payment} = RewardAddress
+  { network: network
+  , payment: payment
+  , delegation: unit
+  }
+
+
+-- | Convert address to its byte representation
+addressBytes
+  :: forall a p d
+   . Address a p d
+  => a
+  -> ByteArray
+addressBytes = ToAddressCsl >>> toCslRep >>> addressBytesImpl
+
+-- | Parse `RewardAddress` from its byte representation
+rewardAddressFromBytes
+  :: forall p
+   . { payment :: CredType p
+     }
+   -> ByteArray
+   -> Maybe (RewardAddress p)
+rewardAddressFromBytes checks bts = do
+  addrCsl <- rewardAddressFromBytesImpl maybeFfiHelper bts
+  headerCheckRewardAddr maybeFfiHelper checks bts addrCsl
+
+-- | Parse `BaseAddress` from its byte representation
+baseAddressFromBytes
+  :: forall p d
+   . { payment :: CredType p
+     , delegation :: CredType d
+     }
+   -> ByteArray
+   -> Maybe (BaseAddress p d)
+baseAddressFromBytes checks bts = do
+  addrCsl <- baseAddressFromBytesImpl maybeFfiHelper bts
```

Comment:
I think it would be nice to have an example showing how the `checks` argument should be used. For example, say I have a `ByteArray` that I want to get a `PubKeyAddress` from. What should I provide as the `checks` argument?

Reply:
PubkeyAdress does not require checks argument, there's a dedicated function that does it for you. I will extend docs for the base function.